### PR TITLE
Se agregaron los ejercicios de Collection

### PR DESCRIPTION
**Pull request by omares465github**
Son los ejercicios correspondientes al 30 de Agosto

### CODE REVIEW

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+package collection;
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:2]: How is this works without imports?!!! Unbelievable!
I don't think you have test this right?

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+public class ArrayList {
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int numberLines = in.nextInt();
+        ArrayList<ArrayList<Integer>> lineas = new ArrayList<>();
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:8]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: OmarEspinosaCastro/30-Agosto/HashTables.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
+package collection;
+
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/HashTables.java:3]: Imports missing

File: OmarEspinosaCastro/30-Agosto/Hashset.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int lines = in.nextInt();
+        String[] left = new String[lines];
+        String[] right = new String[lines];
+        for (int indicePair = 0; indicePair < lines; indicePair++) {
+            left[indicePair] = in.next();
+            right[indicePair] = in.next();
+        }
+        HashSet<String> pairs = new HashSet<>(lines);
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/Hashset.java:14]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: OmarEspinosaCastro/30-Agosto/JavaSort.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+
+package collection;
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/JavaSort.java:2]: imports

File: OmarEspinosaCastro/30-Agosto/JavaSort.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+public class JavaSort {
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int numberStudent  = Integer.parseInt(in.nextLine());
+        ArrayList<Student> studentList = new ArrayList<Student>();
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/JavaSort.java:9]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: OmarEspinosaCastro/30-Agosto/List.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+
+package collection;
+
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/List.java:3]: imports

File: OmarEspinosaCastro/30-Agosto/List.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+public class List {
+    public static void main(String[] args) {
+        Scanner in =new Scanner(System.in);
+        int numberElements = in.nextInt();
+        ArrayList<Integer> lista = new ArrayList();
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/List.java:8]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: OmarEspinosaCastro/30-Agosto/Map.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
+
+package collection;
+
+public class Map {
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int lengthDirectory = in.nextInt();
+        in.nextLine();
+        HashMap<String, Integer> directorio = new HashMap<String, Integer>();
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/Map.java:10]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

directorio?? what's that?? only english please

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
+
+package collection;
+
+public class ResultHashTables {
+    static boolean checkMagazine(String[] magazine, String[] note) {
+        HashMap<String, Integer> magazineWords = new HashMap<String, Integer>();
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/ResultHashTables.java:6]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
+
+package collection;
```
Comment (luisod78) [inline OmarEspinosaCastro/30-Agosto/ResultHashTables.java:2]: imports

**Review by luisod78 — CHANGES_REQUESTED** [review]

File: OmarEspinosaCastro/30-Agosto/ArrayList.java (hunk 1 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ArrayList.java:2]: I run the code, only to remove the components and imports
Sin título

File: OmarEspinosaCastro/30-Agosto/Hashset.java (hunk 4 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/Hashset.java]: Added 'list <>' instead of 'ArrayList <>'

**Review by luisod78 — APPROVED** [review]

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ArrayList {
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int numberLines = in.nextInt();
+        List<List<Integer>> lineas = new ArrayList<>();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:12]: English, improve the name. example: rowListList

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+package collection;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ArrayList {
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int numberLines = in.nextInt();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:11]: lineNumber

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+        List<List<Integer>> lineas = new ArrayList<>();
+        for (int row = 0; row < numberLines; row++) {
+            int columns = in.nextInt();
+            List<Integer> linea = new ArrayList<>();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:15]: improve the  name, English please

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+            for (int column = 0; column < columns; column++) {
+                linea.add(in.nextInt()); 
+            }
+            lineas.add(linea);
+        }
+        int queries = in.nextInt();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:21]: Improve the name

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+        for (int numberQuery = 0; numberQuery < queries; numberQuery++) {
+            int positionX = in.nextInt();
+            int positionY = in.nextInt();
+             List<Integer> linea = lineas.get(positionX-1);
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:25]: improve the name, English

File: OmarEspinosaCastro/30-Agosto/ArrayList.java [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+            if (positionY <= linea.size()) {
+                System.out.println(linea.get(positionY-1));
+            } else {
+                System.out.println("ERROR!");
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ArrayList.java:29]: Clound by a static final variable.

File: OmarEspinosaCastro/30-Agosto/HashTables.java [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+import java.io.*;
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/HashTables.java:4]: Don't group imports.

File: OmarEspinosaCastro/30-Agosto/HashTables.java [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.*;
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/HashTables.java:5]: same

File: OmarEspinosaCastro/30-Agosto/HashTables.java [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+public class HashTables {
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int lengthMagazine = in.nextInt();
+        int lengthNote = in.nextInt();
+        String magazine[] = new String[lengthMagazine];
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/HashTables.java:13]: I believe you need to use HashTable in this hackerrank problem.

File: OmarEspinosaCastro/30-Agosto/JavaSort.java [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+public class JavaSort {
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int numberStudent  = Integer.parseInt(in.nextLine());
+        List<Student> studentList = new ArrayList<Student>();
+        while (numberStudent > 0) {
+            int id = in.nextInt();
+            String name = in.next();
+            double cgpa = in.nextDouble();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/JavaSort.java:19]: cgpa?

File: OmarEspinosaCastro/30-Agosto/JavaSort.java [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+            Student st = new Student(id, name, cgpa);
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/JavaSort.java:20]: st?

File: OmarEspinosaCastro/30-Agosto/List.java [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.List;
+
+public class List {
+    public static void main(String[] args) {
+        Scanner in =new Scanner(System.in);
+        int numberElements = in.nextInt();
+        List<Integer> lista = new ArrayList<>();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/List.java:12]: improve the name and English

File: OmarEspinosaCastro/30-Agosto/List.java [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+        for (int i = 0; i < numberElements; i++) {
+            int valueElement = in.nextInt();
+            lista.add(valueElement);
+        }
+        int queries = in.nextInt();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/List.java:17]: Imporve the name

File: OmarEspinosaCastro/30-Agosto/List.java [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+        for (int numberQuery = 0; numberQuery < queries; numberQuery++) {
+            String querie = in.next();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/List.java:19]: Improve the name

File: OmarEspinosaCastro/30-Agosto/List.java [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+            if (querie.equals("Insert")) {
+                int index = in.nextInt();
+                int value = in.nextInt();
+                lista.add(index, value);
+            }else{
+                int index = in.nextInt();
+                lista.remove(index);
+            }
+        }
+        for (Integer elemento : lista) {
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/List.java:29]: English

File: OmarEspinosaCastro/30-Agosto/List.java [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+            System.out.print(elemento + " ");
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/List.java:30]: Same

File: OmarEspinosaCastro/30-Agosto/Map.java [hunk 28]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Scanner;
+import java.util.Map;
+
+public class Map {
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int lengthDirectory = in.nextInt();
+        in.nextLine();
+        Map<String, Integer> directorio = new HashMap<>();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/Map.java:14]: English and improve the name, example directoryMap

File: OmarEspinosaCastro/30-Agosto/Map.java [hunk 29]
```diff
@@ -0,0 +1,31 @@
...
+        for (int rowDirectory = 0; rowDirectory < lengthDirectory; rowDirectory++) {
+            String nameClave = in.nextLine();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/Map.java:16]: No pocho, keyName

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java [hunk 30]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ResultHashTables {
+    static boolean checkMagazine(String[] magazine, String[] note) {
+        Map<String, Integer> magazineWords = new HashMap<>();
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ResultHashTables.java:9]: Improve the name

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java [hunk 31]
```diff
@@ -0,0 +1,28 @@
...
+        for (int indiceMagazine = 0; indiceMagazine < magazine.length; indiceMagazine++) {
+            Integer wordCount = magazineWords.get(magazine[indiceMagazine]);
+            if (wordCount == null) {
+                magazineWords.put(magazine[indiceMagazine], 1);
+            } else {
+                magazineWords.put(magazine[indiceMagazine], wordCount + 1);
+            }
+        }
+        for (int note_index = 0; note_index < note.length; note_index++) {
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ResultHashTables.java:18]: Don't use _ inside variable names, use LowerCamelCase

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java [hunk 32]
```diff
@@ -0,0 +1,28 @@
...
+            Integer founWord = magazineWords.get(note[note_index]);
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/ResultHashTables.java:19]: foun?

File: OmarEspinosaCastro/30-Agosto/Student.java [hunk 33]
```diff
@@ -0,0 +1,28 @@
+
+package collection;
+
+public class Student {
+
+    private int id;
+    private String fname;
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/Student.java:7]: fname ?

File: OmarEspinosaCastro/30-Agosto/Student.java [hunk 34]
```diff
@@ -0,0 +1,28 @@
...
+    private double cgpa;
```
Comment (globant-bootcamp) [inline OmarEspinosaCastro/30-Agosto/Student.java:8]: cgpa?

**Review by globant-bootcamp — CHANGES_REQUESTED** [review]

File: OmarEspinosaCastro/30-Agosto/ArrayList.java (hunk 12 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ArrayList.java:12]: Variable name was changed

File: OmarEspinosaCastro/30-Agosto/ArrayList.java (hunk 13 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ArrayList.java:11]: Variable name was changed

File: OmarEspinosaCastro/30-Agosto/ArrayList.java (hunk 14 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ArrayList.java:15]: Variable name was changed

File: OmarEspinosaCastro/30-Agosto/ArrayList.java (hunk 15 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ArrayList.java:21]: Variable name was changed

File: OmarEspinosaCastro/30-Agosto/ArrayList.java (hunk 16 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ArrayList.java:25]: Variable name was changed

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java (hunk 30 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ResultHashTables.java:9]: Variable name was changed

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java (hunk 31 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ResultHashTables.java:18]: camel case was added

File: OmarEspinosaCastro/30-Agosto/ResultHashTables.java (hunk 32 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/ResultHashTables.java:19]: Variable name was changed

File: OmarEspinosaCastro/30-Agosto/Student.java (hunk 33 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/Student.java:7]: ariable name was changed

File: OmarEspinosaCastro/30-Agosto/Student.java (hunk 34 above)
Comment (omares465github) [reply OmarEspinosaCastro/30-Agosto/Student.java:8]: ariable name was changed

**Review by globant-bootcamp — APPROVED** [review]

**Pull request merged**